Content.Server/StoreDiscount/Systems/StoreDiscountSystem.cs

```diff
@@ -25,10 +26,51 @@ public override void Initialize()
     {
         base.Initialize();
 
-        SubscribeLocalEvent<StoreInitializedEvent>(OnStoreInitialized);
+        SubscribeLocalEvent<StoreDiscountComponent, ComponentStartup>(OnDiscountStartup);
+        SubscribeLocalEvent<StoreComponent, StoreAddedEvent>(OnStoreStartup);
         SubscribeLocalEvent<StoreBuyFinishedEvent>(OnBuyFinished);
     }
 
+    /// <summary>
+    /// This event is relevant when a PDA/uplink is created and a discount component is already on it
+    /// </summary>
+    private void OnStoreStartup(Entity<StoreComponent> entity, ref StoreAddedEvent args)
+    {
+        if (!TryComp<StoreComponent>(entity, out var storeComponent))
+            return;
+
+        if (!TryComp<StoreDiscountComponent>(entity, out var discountComponent))
+            return;
+
+        ActivateDiscounts(storeComponent, discountComponent);
+    }
```

Comment:
```suggestion
    private void OnStoreStartup(Entity<StoreComponent> entity, ref StoreAddedEvent args)
    {
        if (!TryComp<StoreDiscountComponent>(entity, out var discountComponent))
            return;

        ActivateDiscounts(entity, discountComponent);
    }
```

Oh, sorry, my previous suggestion was a bit bad.
The [`Entity<T>` value contains both the `EntityUid` and the `T` simultaneously](https://github.com/space-wizards/RobustToolbox/blob/7df23e047c222cdaef064952e15c348d19e4f8f7/Robust.Shared/GameObjects/Entity.cs#L13), so the first `TryComp` is unnecessary.

You can explicitly access the `StoreComponent` in this case as `entity.Comp`, but there's an [implicit conversion operator](https://github.com/space-wizards/RobustToolbox/blob/7df23e047c222cdaef064952e15c348d19e4f8f7/Robust.Shared/GameObjects/Entity.cs#L39) which makes it so an `Entity<T>` can be used anywhere a `T` is specifically needed, and the compiler will implicitly insert the conversion for you.